chirp-impl/src/main/java/sample/chirper/chirp/impl/JpaSessionImpl.java

```diff
@@ -55,14 +65,42 @@
         });
     }
 
-    private CompletionStage<Done> close() {
+    private CompletionStage<EntityManagerFactory> createEntityManagerFactory() {
         return executeInSlickContext(() -> {
-            factory.close();
-            return Done.getInstance();
+            log.debug("Initializing JPA EntityManagerFactory with persistence unit name {}", PERSISTENCE_UNIT_NAME);
+            long startTime = System.currentTimeMillis();
+            while (System.currentTimeMillis() - startTime <= INIT_TIMEOUT_MILLIS &&
+                    !Thread.currentThread().isInterrupted()) {
+                try {
+                    return Persistence.createEntityManagerFactory(PERSISTENCE_UNIT_NAME);
+                } catch (Exception e) {
+                    log.warn("Exception while initializing JPA EntityManagerFactory", e);
+                }
+                if (System.currentTimeMillis() - startTime <= INIT_TIMEOUT_MILLIS) {
+                    log.info("Will retry initializing JPA EntityManagerFactory in {} ms", INIT_RETRY_INTERVAL_MILLIS);
+                    try {
+                        Thread.sleep(INIT_RETRY_INTERVAL_MILLIS);
+                    } catch (InterruptedException e) {
+                        Thread.currentThread().interrupt();
+                    }
+                }
+            }
+            log.error("Cancelled initializing JPA EntityManagerFactory after {} ms", INIT_TIMEOUT_MILLIS);
+            throw new CancellationException(
+                    "Cancelled initializing JPA EntityManagerFactory after " + INIT_TIMEOUT_MILLIS + " ms"
+            );
```

Comment:
The above logic is a bit screwed because scheduleOnce doesn't return a future, instead you'll have to create a future and redeem it with the result of the nested tryCreate, but hopefully you get the gist.